library/src/main/java/com/okta/oidc/AuthenticationPayload.java

```diff
@@ -71,6 +73,48 @@ public Builder setState(@Nullable String state) {
             return this;
         }
 
+        /**
+         * Identity provider (default is Okta, unless you are using Social Login or enterprise SAML)
+         * If using a social login provider use this to set the client id of the app.
+         *
+         * @param idp This is the client id of the app social login provider.
+         * @return current Builder
+         * @see "Social Login
+         * <https://developer.okta.com/docs/concepts/social-login/#features>"
+         */
+        public Builder setIdp(@Nullable String idp) {
```

Comment:
It's a parameter in [authorize](https://developer.okta.com/docs/reference/api/oidc/#request-parameters) endpoint.
```bash
A space delimited list of scopes to be provided to the Social Identity Provider when performing Social Login. These scopes are used in addition to the scopes already configured on the Identity Provider.
```

Reply:
The question still the same, why we need this if we configure google client_id and google scopes in dashboard? I suspect these params are ignored by Okta

Reply:
If idp is not in the authorize parameter it will default to Okta. For example you can setup multiple social login providers in your org with different client id but you don't set the scope since it is client specific. Then provide an option to use the specific provider and scope by using the idp/idp_scope parameters like so:

/authorize?idp=&client_id={okta app id}... - Okta default
/authorize?idp={Google client_id}&idp_scope={idp specific}&client_id={okta app id}... - Google
/authorize?idp={facebook client_id}&idp_scope={idp specific}&client_id={okta app id}... - Facebook.

Reply:
For the many configured Idp's it totally makes sense.

What it the result for the following use case:
- You have configured and active Google idp in Okta dashboard
- You don't use idp and idp_scope parameters in oidc request

What is the result? Does Okta automatically redirects to Google?

Reply:
I don't think an option exists to configure routing rules since this is done in the client. In my org I add Google as a provider and don't see any option to configure any rules. But if attempting to login without an idp it will use the okta sign-in flow.

 In the sample test app I have a sign-in (Okta)and social login (Google). Same client but using different IDP one is okta the other is google. If you sign in with google and the user email/profile doesn't exist in Okta, it will automatically create a user account in Okta. This depends on the setting in your org of what information to map from google to okta.

Reply:
Routing rules tab is on the same page where you configure Google Idp. I will show you tomorrow. I've just tested this scenario and sign-in widget automatically redirected to google sign in page without using idp and idp_scope params

However your scenario can work with multiple idp's and disabled routing rule. So this is great use case and addition to the sdk!